fecgov/fec-cms
# Feature/3041 restore datatables pagination

## Summary

- Resolves #3041 
- Change paging style to include page numbers instead of only < and >

## Impacted areas of the application
- Added table information to the dom (tables.js, line 26)
- Changed the pagination format/elements (tables.js, line 448)
- Added documentation for the changes in tables.js
- Adjusted css styles to
  - make disabled buttons (see < on any page 1) appear as disabled as per [style guide](https://fec-pattern-library.app.cloud.gov/components/detail/basic.html)
  - style the ellipses so they're as tall as the page number buttons, so the … is bottom-aligned rather than slightly off vertical middle
  - added space under the "Results per page" pulldown for the smallest widths
  - right-aligned the "Showing…" text for the smallest widths+

## Screenshots
Before/current:
![image](https://user-images.githubusercontent.com/26720877/68873824-3460a500-06ce-11ea-98f2-61d0231bc442.png)

After, smallest:
![image](https://user-images.githubusercontent.com/26720877/68872553-375a9600-06cc-11ea-898d-72a7b6380372.png)
After, wider:
![image](https://user-images.githubusercontent.com/26720877/68872585-46d9df00-06cc-11ea-9d0e-866595b2ce97.png)

## Related PRs
None

## How to test
- Pull the branch, `npm i`, `npm run build`, `./manage.py runserver` like always
- [Candidate details](http://127.0.0.1:8000/data/candidate/P00010298/?tab=raising) raising & spending were already showing "Showing…"; make sure they're unchanged
- [Individual contributions](http://127.0.0.1:8000/data/receipts/individual-contributions/?two_year_transaction_period=2020&two_year_transaction_period=2018&contributor_state=PR&committee_id=C00588392) should have the new paging style
- [Candidates lists](http://127.0.0.1:8000/data/candidates/president/?election_year=2020&cycle=2020&election_full=true) should have the new paging style
- **Make sure other tables across the site are paging as expected**
- _Check various browser widths_

## Reference
Pattern library for [pagination](https://fec-pattern-library.app.cloud.gov/components/detail/feed-pagination.html) and [buttons](https://fec-pattern-library.app.cloud.gov/components/detail/basic.html)

____
